### PR TITLE
IMRPhenomPv2 fix

### PR DESCRIPTION
**Pull request by ravioli1369**

### CODE REVIEW

File: ml4gw/waveforms/cbc/phenom_d.py [hunk 1]
```diff
@@ -384,17 +407,20 @@ def phenom_d_inspiral_amp(
 
         return amp, Damp
 
-    def phenom_d_phase(self, Mf, mass_1, mass_2, eta, eta2, chi1, chi2, xi):
+    def phenom_d_phase(
+        self, Mf, mass_1, mass_2, eta, eta2, chi1, chi2, xi, fRD=None, fDM=None
```
Comment (deepchatterjeeligo) [inline ml4gw/waveforms/cbc/phenom_d.py:411]: Looks good. Just leave a comment here that the `None` values are for phenomd and for phenomp the value need to be supplied.

File: ml4gw/waveforms/cbc/phenom_d.py [hunk 2]
```diff
@@ -454,15 +480,18 @@ def phenom_d_phase(self, Mf, mass_1, mass_2, eta, eta2, chi1, chi2, xi):
...
+    ):
         alpha1 = self.alpha1Fit(eta, eta2, xi)
         alpha2 = self.alpha2Fit(eta, eta2, xi)
         alpha3 = self.alpha3Fit(eta, eta2, xi)
         alpha4 = self.alpha4Fit(eta, eta2, xi)
         alpha5 = self.alpha5Fit(eta, eta2, xi)
 
         # merger ringdown
-        fRD, fDM = self.fring_fdamp(eta, eta2, chi1, chi2)
+        if (fRD is None) or (fDM is None):
```
Comment (deepchatterjeeligo) [inline ml4gw/waveforms/cbc/phenom_d.py:493]: For completeness, throw a RuntimeError if only one of fRD or fDM is not none.

Comment (ravioli1369) [reply ml4gw/waveforms/cbc/phenom_d.py:493]: would be better to make the `or` into an `and`?

Comment (deepchatterjeeligo) [reply ml4gw/waveforms/cbc/phenom_d.py]: Sure, but check the behavior. It's better to be explicit and to raise the error for unexpected combinations.

Comment (ravioli1369) [reply ml4gw/waveforms/cbc/phenom_d.py]: yea I did that in the latest commit, could you take a look?

**Review by deepchatterjeeligo — APPROVED** [review]

File: tests/waveforms/cbc/test_cbc_waveforms.py [hunk 3]
```diff
@@ -409,14 +434,14 @@ def test_phenom_p():
         hc_lal_data = hc_lal.data.data[lal_mask]
 
         assert np.allclose(
-            1e21 * hp_lal_data.real, 1e21 * hp_ml4gw.real.numpy(), atol=2e-3
+            1e21 * hp_lal_data.real, 1e21 * hp_ml4gw.real.numpy(), atol=1e-2
```
Comment (wbenoit26) [inline tests/waveforms/cbc/test_cbc_waveforms.py:437]: Why was it necessary to increase the tolerance? Do we agree less closely with lalsim after these changes? Or are we testing a wider range of parameter space?

Comment (ravioli1369) [reply tests/waveforms/cbc/test_cbc_waveforms.py:437]: The previous version of the waveform didn't match at all with lalsim when the tests were expanded to a larger parameter space. Now we match well in the full space, but certain parameters cause differences larger than ~6e-3, so I made it 1e-2.

File: ml4gw/waveforms/cbc/phenom_p.py [hunk 4]
```diff
@@ -320,42 +328,61 @@ def PhenomPOneFrequency(
...
         )
-        diffRDphase = -diffRDphase[:, 50]
-        # MfRD = torch.outer(M_s, fRD)
-        # Dphase = torch.diag(
-        #     -self.phenom_d_phase(
-        #         MfRD, m1, m2, eta, eta2, chi1, chi2, xi)[1] * M_s
-        # ).view(-1, 1)
+        # interpolate at x = 1, as thats the same as f = fRD
+        diffRDphase = -self.interpolate(
+            torch.tensor([1]), x[1:-1], diffRDphase
```
Comment (AndyC80297) [inline ml4gw/waveforms/cbc/phenom_p.py:384]: The x[1:-1] and diffRDphase will have different shapes when the batch size is not one. 
Probably need to let x[1:-1] follow the shape of diffRDphase so the function can correctly interpolate.

Comment (EthanMarx) [reply ml4gw/waveforms/cbc/phenom_p.py:384]: @ravioli1369 can you look into this?

Comment (EthanMarx) [reply ml4gw/waveforms/cbc/phenom_p.py:384]: We should also add to the tests a test that uses batched inputs

Comment (deepchatterjeeligo) [reply ml4gw/waveforms/cbc/phenom_p.py:384]: Ok this is a good catch. They TaylorF2 and PhenomD tests had the same parameters repeated along the batch dimension. I think that is the easiest extension to the tests to ensure the plumbing is working as expected.

Comment (deepchatterjeeligo) [reply ml4gw/waveforms/cbc/phenom_p.py:384]: Sorry for the oversight during the review.

**Pull request merged**